cdap-source-control/src/main/java/io/cdap/cdap/sourcecontrol/worker/TaskUtil.java

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright © 2023 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.cdap.sourcecontrol.worker;
+
+import com.google.inject.Guice;
+import com.google.inject.Injector;
+import io.cdap.cdap.common.conf.CConfiguration;
+import io.cdap.cdap.common.guice.ConfigModule;
+import io.cdap.cdap.common.guice.RemoteAuthenticatorModules;
+import io.cdap.cdap.common.internal.remote.RunnableTaskModule;
+import io.cdap.cdap.security.auth.context.AuthenticationContextModules;
+import io.cdap.cdap.security.guice.SecureStoreClientModule;
+import org.apache.twill.discovery.DiscoveryService;
+import org.apache.twill.discovery.DiscoveryServiceClient;
+
+/**
+ * The utility class that assists source control management tasks.
+ */
+public class TaskUtil {
```

Comment:
Please follow utility class pattern (final, private constructor)

Reply:
try to avoid having util classes if possible, they tend to accumulate unrelated logic over time and encourage poor class design.

In this particular case, maybe have an abstract SourceControlTask class that creates the InMemorySourceControlOperationRunner?

Reply:
Will try the abstract class approach. But I think the biggest blocker is how to mock in the tests

Reply:
Updated to the SourceControlTask class approach, pls have a second round of review, thanks! @tivv @albertshau